code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -231,9 +231,7 @@ Auto Patrol: []"},
 	playsound(loc, 'sound/weapons/cablecuff.ogg', 30, 1, -2)
 	C.visible_message("<span class='danger'>[src] is trying to put zipties on [C]!</span>",\
 						"<span class='userdanger'>[src] is trying to put zipties on you!</span>")
-	spawn(60)
-		if( !Adjacent(C) || !isturf(C.loc) ) //if he's in a closet or not adjacent, we cancel cuffing.
-			return
+	if(do_after(src, 60, target = C))
```

Comment:
Make sure this doesn't make the controller hang.

Reply:
Tested it a few times. No problems with it. No runtimes on my tests

Reply:
There wouldn't be runtimes if it held up the controller.

do_after has a sleep in it. You'd have to keep an eye on whatever loop this proc gets called in.

Then make the `do_after` something obscenely long (like 600) and take note of what happens to the controller and how many times it fires.

if the controller fires every 2 seconds, but suddenly stops firing when beepsky goes to cuff---you have a problem.

Reply:
I can see the problem arising here yeah... Unsure if there is a good way to fix it with the current tools

Reply:
I now use INVOKE_SYNC. Unsure if this is actually async since i could not find great documentation about this.

Reply:
That might not be necessary if `do_after` doesn't hang the controllers.

Just need to make sure it does for sure or not.

Reply:
It doesnt hang it but if this is async then itll be better. The less sleeps the better on the main loops

Reply:
not if a loop has `set waitfor = FALSE`